### PR TITLE
Feature/163187480/enable login google

### PR DESCRIPTION
**Pull request by rajeman**
#### What does this PR do?
- It ensures google can be used to signup or login to the app
#### Description of Task to be completed?
- [x] setup google authentication
- [x] ensure an account is created in the app for new users with google
- [x] refactor User model to allow null input for username and password 
#### How should this be manually tested?
- ensure Google developer ID and Secret is set
- start the app
- navigate to `/api/v1/users/google` using a **browser** 
- you should be redirected to google
- login with valid google account 
- server will respond with a JWT token and user details

#### What are the relevant pivotal tracker stories?
- [#1631874800](https://www.pivotaltracker.com/n/projects/2229853/stories/163187480)

### CODE REVIEW

File: config/strategies/googleStrategy.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import passport from 'passport';
+import { OAuth2Strategy as GoogleStrategy } from 'passport-google-oauth';
+import Users from '../../controllers/user';
```
Comment (oreoluwade) [inline config/strategies/googleStrategy.js]: Could you import the `Users` controller directly from the controllers directory?

Comment (rajeman) [reply config/strategies/googleStrategy.js]: I will correct that

File: config/strategies/googleStrategy.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  init() {
+    passport.use(new GoogleStrategy({
+      clientID: process.env.GOOGLE_ID,
+      clientSecret: process.env.GOOGLE_SECRET,
+      callbackURL: '/api/v1/users/google/redirect'
+    },
+    (accessToken, refreshToken, profile, done) => {
+      done(null, {
+        email: profile.emails[0].value,
+        firstName: profile.name.givenName,
```
Comment (oreoluwade) [inline config/strategies/googleStrategy.js]: Are these details used/saved to the DB: `firstName` and `lastName`?

Comment (rajeman) [reply config/strategies/googleStrategy.js]: No.

Comment (rajeman) [reply config/strategies/googleStrategy.js]: They will probably be needed in the profile table

Comment (oreoluwade) [reply config/strategies/googleStrategy.js]: Okay. We can take it out for now then. Will revisit when we need to add them to the profile table.

Comment (rajeman) [reply config/strategies/googleStrategy.js]: Alright

**Review by oreoluwade — APPROVED** [review]

**Pull request merged**